src/Control/Carrier/FossaApiClient/Internal/Core.hs

```diff
@@ -102,3 +119,35 @@ getIssues ::
 getIssues rev = do
   apiOpts <- ask
   API.getIssues apiOpts rev
+
+getSignedUploadUrl ::
+  ( Has (Lift IO) sig m
+  , Has Diagnostics sig m
+  , Has (Reader ApiOpts) sig m
+  ) =>
+  ProjectRevision ->
+  m SignedURL
+getSignedUploadUrl ProjectRevision{..} = do
+  apiOpts <- ask
+  API.getSignedURL apiOpts projectRevision projectName
```

Comment:
Can we please give this its own type? Archive uploads never have a `projectBranch` associated, which means that any time we want to use this the value provided to the `ProjectRevision` instance is silently ignored.

While I recognize it's convenient to use the type since it mostly fits, I think it's important to make sure APIs use types that reflect what they actually _need_.

Reply:
Okay, I wasn't sure, and the previous code used `Text` for both, which was awkward.  I'll just `newtype` them or something.

Reply:
Yeah! I think newtype wrappers or a new type that is just name and revision would work great!

Reply:
Done.